### PR TITLE
fsm client

### PR DESCRIPTION
**Pull request by sclasen**
/cc @ryanbrainard

### CODE REVIEW

File: fsm/client.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	if err != nil {
+		if ae, ok := err.(aws.APIError); ok {
+			log.Printf("component=client fn=GetState at=list-open error-type=%s message=%s", ae.Type, ae.Message)
+		} else {
+			log.Printf("component=client fn=GetState at=list-open error=%s", err)
+		}
+		return "", nil, err
+	}
+
+	if len(open.ExecutionInfos) > 0 {
```
Comment (ryanbrainard) [inline fsm/client.go]: would it ever be `> 1`? i.e. should this just be `== 1` ?

File: fsm/client.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		execution = open.ExecutionInfos[0].Execution
+	} else {
+		closed, err := c.c.ListOpenWorkflowExecutions(&swf.ListOpenWorkflowExecutionsInput{
```
Comment (ryanbrainard) [inline fsm/client.go]: should this be `ListClosedWorkflowExecutions`?

File: fsm/client.go (hunk 1 above)
Comment (sclasen) [inline fsm/client.go]: oh for open ones I suppose not, thanks.

File: fsm/client.go (hunk 2 above)
Comment (ryanbrainard) [inline fsm/client.go]: can it share the params w/ the similar looking call above?

Comment (sclasen) [inline fsm/client.go]: Yep should be closed and no, it needs to be a ListClosedWorkflowExecutionsInput though the fields are identical :dizzy_face:

File: fsm/client.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		if err != nil {
+			if ae, ok := err.(aws.APIError); ok {
+				log.Printf("component=client fn=GetState at=list-closed error-type=%s message=%s", ae.Type, ae.Message)
+			} else {
+				log.Printf("component=client fn=GetState at=list-closed error=%s", err)
+			}
+			return "", nil, err
+		}
+
+		if len(closed.ExecutionInfos) > 0 {
```
Comment (ryanbrainard) [inline fsm/client.go]: same here about `== 1`

Comment (sclasen) [inline fsm/client.go]: in the general case there can be more than 1, using a page size of 1 should prevent that I suppose.

**Comment by sclasen** [comment]
closes #11

**Pull request merged**